File: 050 Pow(x, n)/pow_iterative.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 0:
            return 1.0
        if n < 0:
            x = 1 / x
            n = -n
        res = 1
        curr_product = x
        while (n != 0):
            if n % 2 == 1:
                res = res * curr_product
            curr_product = curr_product * curr_product
            n = n // 2
        return res

File: 050 Pow(x, n)/Pow_Iterative.java
class Solution {
    public double myPow(double x, int n) {
        long N = n;
        if (N < 0) {
            N = -N;
            x = 1 / x;
        }
        
        double res = 1;
        double curr = x;
        while (N > 0) {
            if (N % 2 == 1) {
                res *= curr;
            }
            curr *= curr;
            N /= 2;
        }
        return res;
    }
}